### PR TITLE
docs(config): add clarification for gitAuthor format to avoid obscure errors during commits.

### PR DESCRIPTION
**Pull request by slonim-greg**
## Changes
Add documentation clarification for `gitAuthor` config option.
Closes #23538 

## Context

The context for this change is documented in [Issue 23538](https://github.com/renovatebot/renovate/discussions/23538).
#### Bottom line:
If one supplies `gitAuthor` or overrides `RENOVATE_GIT_AUTHOR `, it is often important to supply both NAME and EMAIL according to RFC-5322.

## Documentation (please check one with an [x])

- [x] I have updated the documentation

## How I've tested my work (please select one)

I have verified these changes via:

- [ ] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [x] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

**Review by rarkins — COMMENTED** [review]
I'm not sure every platform requires both name and email

**Comment by slonim-greg** [comment]
It would not hurt to include, however, to avoid similar problems, even when a platform does not require it, right?

**Comment by rarkins** [comment]
I prefer not to tell something is required if it's only required for edge cases or a subset of platforms

**Comment by slonim-greg** [comment]
It was a problem for our repository; Moreover, the default setting in the "standard issue" Renovate Runner repo has the name [there](https://gitlab.com/renovate-bot/renovate-runner/-/blob/main/.gitlab-ci.yml#L6). 
We can change the docs language to say that if one encounters the problem I saw, this is what they should do. 
In reality though, this is a pretty low hanging fruit and seems pretty innocuous to include if one overrides this config value. 
Thoughts?

File: docs/usage/configuration-options.md [hunk 1]
```diff
@@ -1105,7 +1105,11 @@ You can't use other filenames because Renovate only checks the default filename
 ## gitAuthor
 
 You can customize the Git author that's used whenever Renovate creates a commit.
-The `gitAuthor` option accepts a RFC5322-compliant string.
+The `gitAuthor` option accepts a [RFC5322](https://datatracker.ietf.org/doc/html/rfc5322)-compliant string.
+Specifically, make sure to include a name followed by an email address, e.g.
```
Comment (rarkins) [inline docs/usage/configuration-options.md:1109]:
```suggestion
It's recommended to include a name followed by an email address, e.g.
```

**Review by rarkins — CHANGES_REQUESTED** [review]

**Review by rarkins — DISMISSED** [review]

**Review by rarkins — DISMISSED** [review]

**Review by rarkins — APPROVED** [review]

**Pull request merged**